industrial_ci/src/tests/source_tests.sh

```diff
@@ -26,6 +26,27 @@ function install_catkin_lint {
     ici_install_pypi_pkgs_for_command catkin_lint "catkin-lint"
 }
 
+function clang_tidy_filter {
+    local source_dir=$1
+    local base_ref=$2
+    local -n _result=$3
+
+    pushd "$source_dir" > /dev/null || exit
+    if [ "$(git rev-parse --abbrev-ref HEAD)" == "$base_ref" ] ; then
+        _result=1
+    else
+        git fetch -q origin "$base_ref"
+        git branch -qf "$base_ref" FETCH_HEAD
+
+        if [ -n "$(git diff --name-only --diff-filter=MA "${base_ref}"..HEAD .)" ]; then
+            eval $_result=1
+        else
+            eval $_result=0
+        fi
+    fi
+    popd > /dev/null || exit
+}
```

Comment:
Why returning the result by a named reference? Better use simple return value like [here](https://github.com/ros-planning/moveit_ci/blob/6ed1a4e19e4be315aa0de227145cd84620d52d4f/util.sh#L294-L302).
```bash
function clang_tidy_filter {
    local source_dir=$1
    local base_ref=$2
    local result

    pushd "$source_dir" > /dev/null || exit
    if [ "$(git rev-parse --abbrev-ref HEAD)" == "$base_ref" ] ; then
        result=0  # HEAD is on par with $base_ref: always perform clang-tidy
    else
        git fetch -q origin "$base_ref"
        git diff --quiet FETCH_HEAD..HEAD .
        result=$?
    fi
    popd > /dev/null || exit
    return $result
}
```